dns_server/src/server/mod.rs

```diff
@@ -0,0 +1,286 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! # Mintlayer DNS seed server
+
+use std::{
+    collections::BTreeMap,
+    net::{IpAddr, Ipv4Addr, Ipv6Addr, SocketAddr},
+    str::FromStr,
+    sync::{
+        atomic::{AtomicU32, Ordering},
+        Arc, Mutex,
+    },
+};
+
+use crypto::random::{make_pseudo_rng, SliceRandom};
+use tokio::{net::UdpSocket, sync::mpsc};
+use trust_dns_client::rr::{rdata::SOA, LowerName, Name, RData, RecordSet, RecordType, RrKey};
+use trust_dns_server::{
+    authority::{
+        AuthLookup, Authority, Catalog, LookupError, LookupOptions, MessageRequest, UpdateResult,
+        ZoneType,
+    },
+    server::RequestInfo,
+    store::in_memory::InMemoryAuthority,
+    ServerFuture,
+};
+
+use crate::{config::DnsServerConfig, error::DnsServerError};
+
+#[derive(Debug)]
+pub enum ServerCommands {
+    AddAddress(IpAddr),
+
+    DelAddress(IpAddr),
+}
+
+pub struct Server {
```

Comment:
Why do we call this Server, and not DnsServer? I would prefer the latter unless there's a good reason not to do it.

Reply:
Sure, updated.